cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -680,6 +683,7 @@ private void startExecution(LoadMonitor loadMonitor,
     _executionStoppedByUser.set(false);
     sanityCheckOngoingMovement();
     _hasOngoingExecution = true;
+    _noOngoingExecutionSemaphore.acquireUninterruptibly();
     _anomalyDetectorManager.maybeClearOngoingAnomalyDetectionTimeMs();
     _anomalyDetectorManager.resetHasUnfixableGoals();
     _stopSignal.set(NO_STOP_EXECUTION);
```

Comment:
If the `Executor` thread context switch right before this line, line 825 `stopExecution()` will set `_stopSignal ` to `STOP`. When switch back, this thread will go on to set `_stopSignal ` to `NO_STOP`,  hence the execution will go through.